### ISSUE TITLE
Unable to Deserialize fields with prefirx "@"

### ISSUE THREAD

**Issue by vikasNew** [JamesNK/Newtonsoft.Json#2153]
Hello,

i am facing a strange issue in Deserializing the json string to Model.

i hava a josn string which has some fields like "@timestamp". They has @ prefix. 

when i deserialize them with "JsonConvert.DeserializeObject" then i could not get the json field value to my model. This is the issue with only fields which has @ prefix.

### Source Code to Deserialize
```
 string responseJson = "{\"@timestamp\":\"2019-09-06T12:40:22.7305343+05:30\",\"level\":\"Information\",\"messageTemplate\":\"test message template\",\"message\":\"test message\"}";
            
            var deserializeResponse = JsonConvert.DeserializeObject<EntityModel>(responseJson, new JsonSerializerSettings
            {
                NullValueHandling = NullValueHandling.Ignore,
                MissingMemberHandling = MissingMemberHandling.Ignore
            });
           
			Console.WriteLine(deserializeResponse.level);
			Console.WriteLine(deserializeResponse.@timestamp.ToString());
```

### Model

```
  public class EntityModel
  {
	  public DateTime @timestamp { get; set; }
	  public string level { get; set; }
	  public string messageTemplate { get; set; }
	  public string message { get; set; }
  }
```
### Output

01/01/0001 00:00:00

Then how to get the right @timestamp value in model ?

Thanks in anticipation !!

**Comment by bartelink** [comment]
The `@` in your code is being interpreted as [a verbatim identifier](http://irisclasson.com/2012/10/30/stupid-question-70-verbatim-identifiers-in-c-should-they-be-used-and-when/) indicator.

The best way to bind is likely to tag it with `[JsonProperty("@timestamp")]` (though the verbatim literal mechanism will also let you name it that way via something like: `public DateTime @"@timestamp" {get;set;}`)

**Comment by dovisutu** [comment]
That's right, and this should only happen in c# since we use [...] as a verbatim identifier indicator in vbnet.(What do f# use for it?)
It's not the question on Json.net, but *by design* in c#.

**Comment by bartelink** [comment]
@dovisutu int F#, you put double ticks around verbatim identifiers. e.g. 

    ```fsharp
    let [<Fact>] ``This is a test method`` () = Assert.Equal(1,1)
    ```
@vikasNew Can you close this please? (unless you have an open problem)
